plugins/common/proxy/proxy.go

```diff
@@ -22,3 +24,25 @@ func (p *HTTPProxy) Proxy() (proxyFunc, error) {
 	}
 	return http.ProxyFromEnvironment, nil
 }
+
+type TCPProxy struct {
+	ProxyURL string `toml:"proxy_url"`
+}
+
+func (p *TCPProxy) Proxy() (*ProxiedDialer, error) {
+	var dialer proxy.Dialer
+	if len(p.ProxyURL) > 0 {
+		parsed, err := url.Parse(p.ProxyURL)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing proxy url %q: %w", p.ProxyURL, err)
+		}
+
+		if dialer, err = proxy.FromURL(parsed, proxy.Direct); err != nil {
+			return nil, err
+		}
+	} else {
+		dialer = proxy.FromEnvironment()
+	}
```

Comment:
If the user doesn't set the proxy url, I would expect not to use a proxy at all, not to use the system proxy settings.

Does proxy.FromEnvironment() not use a proxy in this case or does it use the system proxy settings?

Maybe we need two settings: proxy_url for the url, and something like use_proxy (bool). If use_proxy is true but proxy_url isn't set, then use the system proxy.

Reply:
Having worked on a corporate network that required a proxy for some things, but not all, I prefer that applications only use the proxy when explicitly told to.